cep-9999.md

```diff
@@ -0,0 +1,96 @@
+
+Virtual package names MUST only consist of lowercase ASCII letters, numbers, hyphens, periods and underscores. They MUST NOT use two consecutive separators, with one exception: they MUST start with two underscores.
+
+This means that distributable package names MUST match the following case-insensitive regex: `^(([a-z0-9])|([a-z0-9_](?!_)))[._-]?([a-z0-9]+(\.|-|_|$))*$`.
+
+Virtual package names MUST follow this other regex: `^__[a-z0-9][._-]?([a-z0-9]+(\.|-|_|$))*$`.
+
+In all cases, the maximum length of a package name MUST NOT exceed 128 characters.
+
+### Version strings
+
+Version strings MUST only consist of digits, periods, lowercase ASCII letters, underscores, plus
+symbols, and exclamation marks. Additional rules apply but are out of scope in this CEP and will be
+discussed separately.
+
+The maximum length of a version string MUST NOT exceed 128 characters.
+
+### Build strings
+
+Builds strings MUST only consist of ASCII letters, numbers, periods, plus symbols, and underscores. They MUST match this regex `^[a-zA-Z0-9_\.+]+$`.
+
+The maximum length of a build string MUST NOT exceed 128 characters.
+
+### Filenames
+
+Regular conda artifacts MUST be named following this scheme:
+
+```
+<package name>-<version string>-<build string>.<extension>
+```
+
+Virtual conda packages SHOULD NOT need filename standardization.
+
+### Channels
+
+A conda channel is defined as a location where one can find one or more `repodata.json` files arranged in one subdirectory (_subdir_) each. `noarch/repodata.json` MUST be present to consider the parent location a channel.
+
+#### Channel names
+
+With that definition in mind, a channel name is understood as the path component of the URI leading to the `noarch/repodata.json` location, without leading or trailing slashes. For example, given `https://conda.anaconda.org/conda-forge/noarch/repodata.json`, the full path component is `conda-forge/noarch/repodata.json`. The part leading to `noarch/repodata.json` is thus `conda-forge`, the channel name. For local repodata such as `file:///home/username/channel/noarch/repodata.json`, the channel name is `home/username/channel`.
+
+Channel names MUST only consist of lowercase ASCII letters, numbers, hyphens, periods, slashes and underscores. They MUST start and end with a letter or a digit. They MUST NOT contain two consecutive separators (hyphen, underscore, period, slash). In other words, they MUST match this regex: `^[a-z0-9]+((-|_|.)[a-z0-9]+)*$`.
+
+The maximum length of a channel name SHOULD NOT exceed 128 characters.
+
+#### Subdir names
+
+Channel subdir names MUST only consist of lowercase ASCII letters, numbers, and hyphens. They MUST start and end with a letter or a number. They MUST NOT contain two consecutive hyphens. They MUST match this regex: `^[a-z0-9]+(-[a-z0-9]+)*$`.
```

Comment:
I think we should constrain this even more and say the subdir name must either:

* Be the string literal `noarch`; or
* Follow the pattern `{os}-{architecture}`, where each of `{os}` and `{architecture}` MUST contain only lowercase ASCII letters or numbers (`[a-z0-9]`).

I'm not sure we gain anything by allowing more than one hyphen in the subdir name; the regexp you have here (`^[a-z0-9]+(-[a-z0-9]+)*$`) allows, e.g., `linux-or-vms-on-x86-64-or-vax` as a subdir name.